### PR TITLE
Move encoders to the encoding submodule

### PR DESCRIPTION
**Pull request by k-chaney**
Moved the encoders into the encoding submodule and organized the submodule's components.

Split the encoding submodule into three parts
1) encodings - contains the underlying functional units
2) loaders - lazy loaders that receive an iterator of data and then apply the encoding
3) encoders - classes that are callable and store set parameters for each encoding

### CODE REVIEW

File: bindsnet/encoding/encoders.py [hunk 1]
```diff
@@ -86,30 +76,30 @@ def __init__(self, time: int, dt: float = 1.0, **kwargs):
...
-        self.enc = encoding.poisson
+        self.enc = encodings.poisson
 
 
 class RankOrderEncoder(Encoder):
     def __init__(self, time: int, dt: float = 1.0, **kwargs):
         """
         Creates a callable RankOrderEncoder which encodes as defined in
-        :code:`bindsnet.encodings.rank_order`
+        :code:`bindsnet.encodingss.rank_order`
```
Comment (djsaunde) [inline bindsnet/encoding/encoders.py]: Typo.

File: bindsnet/encoding/encoders.py [hunk 2]
```diff
@@ -86,30 +76,30 @@ def __init__(self, time: int, dt: float = 1.0, **kwargs):
...
-        self.enc = encoding.bernoulli
+        self.enc = encodings.bernoulli
 
 
 class PoissonEncoder(Encoder):
     def __init__(self, time: int, dt: float = 1.0, **kwargs):
         """
         Creates a callable PoissonEncoder which encodes as defined in
-        :code:`bindsnet.encodings.poisson`
+        :code:`bindsnet.encodingss.poisson`
```
Comment (djsaunde) [inline bindsnet/encoding/encoders.py]: Typo.

File: bindsnet/encoding/encoders.py [hunk 3]
```diff
@@ -49,34 +39,34 @@ class SingleEncoder(Encoder):
...
-        self.enc = encoding.repeat
+        self.enc = encodings.repeat
 
 
 class BernoulliEncoder(Encoder):
     def __init__(self, time: int, dt: float = 1.0, **kwargs):
         """
         Creates a callable BernoulliEncoder which encodes as defined in
-        :code:`bindsnet.encodings.bernoulli`
+        :code:`bindsnet.encodingss.bernoulli`
```
Comment (djsaunde) [inline bindsnet/encoding/encoders.py]: Typo.

File: bindsnet/encoding/encoders.py [hunk 4]
```diff
@@ -49,34 +39,34 @@ class SingleEncoder(Encoder):
...
-        self.enc = encoding.single
+        self.enc = encodings.single
 
 
 class RepeatEncoder(Encoder):
     def __init__(self, time: int, dt: float = 1.0, **kwargs):
         """
         Creates a callable RepeatEncoder which encodes as defined in
-        :code:`bindsnet.encodings.repeat`
+        :code:`bindsnet.encodingss.repeat`
```
Comment (djsaunde) [inline bindsnet/encoding/encoders.py]: Typo.

File: bindsnet/encoding/encoders.py [hunk 5]
```diff
@@ -49,34 +39,34 @@ class SingleEncoder(Encoder):
     def __init__(self, time: int, dt: float = 1.0, sparsity: float = 0.3, **kwargs):
         """
         Creates a callable SingleEncoder which encodes as defined in
-        :code:`bindsnet.encodings.single`
+        :code:`bindsnet.encodingss.single`
```
Comment (djsaunde) [inline bindsnet/encoding/encoders.py]: Typo.

**Comment by Hananel-Hazan** [comment]
Thanks @k-chaney good work!

**Pull request merged**